azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -389,3 +390,21 @@ def test_empty_result(self, mock_post):
         query = """print 'a' | take 0"""
         response = client.execute_query("PythonTest", query)
         self.assertTrue(response.primary_results[0])
+
+    @pytest.mark.skipif(sys.version_info < (3, 6), reason="requires python3.6 or higher")
+    def test_running_in_parallel(self):
+        """authenticated clients should be able to do things in parallel"""
+        from multiprocessing import Pool
+        cluster = "https://somecluster.kusto.windows.net"
+        client = KustoClient(cluster)
+        inputs = [client for i in range(5)]
+        with Pool(2) as p:
+            responses = p.map(getendpoint, inputs)
```

Comment:
@danield137, this basically covers it -- the api we're trying to call in our app is KustoClient.execute (which appears to turn around and use requests). 
This fix just lets the client object become accessible from the pool which implicitly uses pickle for transport.